docs/caching2.md

```diff
@@ -0,0 +1,100 @@
+
+There is also an escape hatch - a query param in the URL: `?nocache=true` - that will force the CDN to bypass the cache and hit Metaphysics directly _despite_ a `@cacheable` directive. This can be useful for a privileged user to see an unpublished page (which we wouldn't want to cache).
+
+Setting/unsetting the `Cache-Control` header, and propagating any `max-age` is accomplished in a Relay middleware: [`System/Relay/middleware/cacheHeaderMiddleware.tsx`](System/Relay/middleware/cacheHeaderMiddleware.tsx).
+
+Un-setting the access token header for `@cacheable` queries is accomplished in a Relay middleware: [`System/Relay/createRelaySSREnvironment.ts`](System/Relay/createRelaySSREnvironment.ts).
+
+#### Logged out:
+
+By default, we typically cache all queries, unless otherwise configured with `cacheConfig: { force: true }` or `serverCacheTTL: 0`. How this works is:
+
+- User visits a page and Relay issues a GraphQL request.
+- Unless otherwise specified - this request will not contain an access token header _or_ a `no-cache` header value and so can be cached by the CDN (with an optional `max-age`).
+- As user navigates on the client in SPA mode, queries continue to be issued that can be cached by the CDN (unless otherwise specified).
+
+### Configuring Cache
+
+At the router level, we can configure things in a few different ways:
+
+1. Never cache, period:
+
+```tsx
+const myRouter = [
+  {
+    path: "/some-app",
+    // Tell relay network layer to never cache. This flows down to setting
+    // a `no-cache` header when making the CDN request, thus bypassing that cache as well.
+    cacheConfig: {
+      force: true,
+    },
+  },
+]
+```
+
+2. When logged out, never cache on the server, always cache on the client. This ensure users will always receive fresh data when they first visit a page, but once visited, page will be cached on the client for a given session. (Timely pages like `/auction/:id` apply this strategy.)
+
+```tsx
+const myRouter = [
+  {
+    path: "/some-app",
+    // Tell redis to never cache queries on the server. We still cache on
+    // client-side SPA transitions at the relay level (only during the session).
+    // Only applies while logged out!
+    // This flows down to setting a `no-cache` header when making the CDN request, thus bypassing that cache as well.
+    serverCacheTTL: 0,
+  },
+]
+```
+
+3. Configure the server-side TTL cache time for a given route:
+
+```tsx
+const myRouter = [
+  {
+    path: "/some-app",
+    // This flows down to a `max-age` value in the CDN request as well.
+    serverCacheTTL: 24 * 60 * 60, // 24 hours,
+  },
+]
+```
+
+By default, the value of `serverCacheTTL` falls back to our `GRAPHQL_CACHE_TTL` env config, typically set to 1 hour.
```

Comment:
This no longer applies -- might be worth mentioning how we have default fallbacks in the worker code in MP and perhaps link over to that line

Reply:
I'll update to point out the diff, but that ENV var `GRAPHQL_CACHE_TTL` _is_ still (optionally as there's a default) used for configuration for the Relay cache in https://github.com/artsy/force/blob/69a16ffca38d72e4cbaa604a56fbcb6ba18901e9/src/System/Relay/createRelaySSREnvironment.ts#L126

Reply:
Ah yah, i just mean the `serverCacheTTL` part. That's configured in the worker now. So if we keep it, we can update to mention that it only applies to the client